brilliant-builders/firestarter/game_engine/object.py

```diff
@@ -0,0 +1,282 @@
+
+        self.death_sound = death_sound
+
+    def set_lives(self, value: int) -> None:
+        """Set the players lives."""
+        self.lives = value
+
+    def on_cam_move(self, offset: Tuple[float, float]) -> None:
+        super().on_cam_move(offset)
+        self.death_threshold += offset[1]
+        self.respawn = (self.respawn[0] + offset[0], self.respawn[1] + offset[1])
+        # self.checkpoint = (self.checkpoint[0] + offset[0], self.checkpoint[1] + offset[1])
+
+    def update(self, other_sprites: List[Sprite]) -> None:
+        """Update the players position and handle collisions (very inefficiently!!)"""
+        # update the position
+        old_pos = (self.pos[0], self.pos[1])
+
+        # update our velocity
+        self.vel = (self.vel_x + self.acc_x, self.vel_y + self.acc_y)
+
+        # first try to apply up/downwards velocity
+        self.pos = (self.pos[0] + self.vel_x, self.pos[1] + self.vel_y)
+        if self.collides_with(other_sprites):  # we are colliding in x or y
+
+            self.pos = (old_pos[0], old_pos[1])
+            self.pos = (self.pos[0] + self.vel_x, self.pos[1])
+
+            if self.collides_with(other_sprites):  # colliding in x
+                self.vel_x = 0
+                self.pos = (old_pos[0], old_pos[1])
+                if not self.wall_boost:
+                    self.is_standing = False
+
+            old_pos = (self.pos[0], self.pos[1])
+            self.pos = (old_pos[0], old_pos[1] + self.vel_y)
+
+            if self.collides_with(other_sprites):  # colliding in y
+                self.vel_y = 0
+                self.pos = (old_pos[0], old_pos[1])
+                # this also tells us we are standing on something!
+                self.is_standing = True
+        else:
+            self.is_standing = False
+
+        # dampen the velocity to come to a stop
+        if self.is_standing:
+            self.vel_x *= .25
+            self.vel_y *= .5
+        else:
+            self.vel_x *= .75
+            self.vel_y *= .90
+        # apply some downwards acceleration (gravity)
+        self.acc_y = -1
+
+        if self.pos[1] < self.death_threshold:
+            # we fell out of the map!
+            self.lives -= 1
+            if self.death_sound:
+                self.death_sound.play()
+            if self.lives > 0:
+                # respawn at the checkpoint
+                if self.checkpoint:
+                    self.pos = (self.checkpoint.pos[0], self.checkpoint.pos[1] + 70)
+                else:
+                    self.pos = self.respawn
+            else:
+                self.checkpoint.deactivate()
+                self.checkpoint = None
+                self.pos = self.respawn
+
+    def collides_with(self, other_sprites: List[Sprite]) -> bool:
+        # deal with collisions
+        for sprite in other_sprites:
+            if sprite != self and self.collide_widget(sprite):
+                if sprite.on_collision(self):
+                    return True
```

Comment:
This method will return either `None` or `True`, not `False` or `True` as the type hint suggests.
You should add a `return False` after your for loop to ensure that the method returns a bool.